packages/blttest/package.py

```diff
@@ -0,0 +1,272 @@
+    # -----------------------------------------------------------------------
+    # Basics
+    depends_on("cmake@3.8.2:", type='build')
+    depends_on("mpi", when="+mpi")
+
+    def flag_handler(self, name, flags):
+        if self.spec.satisfies('%cce') and name == 'fflags':
+            flags.append('-ef')
+
+        if name in ('cflags', 'cxxflags', 'cppflags', 'fflags'):
+            return (None, None, None)  # handled in the cmake cache
+        return (flags, None, None)
+
+    def _get_sys_type(self, spec):
+        sys_type = spec.architecture
+        # if on llnl systems, we can use the SYS_TYPE
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+        return sys_type
+
+    @property
+    def cache_name(self):
+        hostname = socket.gethostname()
+        if "SYS_TYPE" in env:
+            # Are we on a LLNL system then strip node number
+            hostname = hostname.rstrip('1234567890')
+        return "{0}-{1}-{2}@{3}.cmake".format(
+            hostname,
+            self._get_sys_type(self.spec),
+            self.spec.compiler.name,
+            self.spec.compiler.version
+        )
+
+    def initconfig_compiler_entries(self):
+        spec = self.spec
+        entries = super(Blttest, self).initconfig_compiler_entries()
+
+        if "+fortran" in spec or self.compiler.fc is not None:
+            entries.append(cmake_cache_option("ENABLE_FORTRAN", True))
+        else:
+            entries.append(cmake_cache_option("ENABLE_FORTRAN", False))
+
+        if ((self.compiler.fc is not None)
+           and ("gfortran" in self.compiler.fc)
+           and ("clang" in self.compiler.cxx)):
+            libdir = pjoin(os.path.dirname(
+                           os.path.dirname(self.compiler.cxx)), "lib")
+            flags = ""
+            for _libpath in [libdir, libdir + "64"]:
+                if os.path.exists(_libpath):
+                    flags += " -Wl,-rpath,{0}".format(_libpath)
+            description = ("Adds a missing libstdc++ rpath")
+            if flags:
+                entries.append(cmake_cache_string("BLT_EXE_LINKER_FLAGS", flags,
+                                                  description))
+
+        # TODO: reenable when multi-variable variant is made
+        # if "+cpp14" in spec:
+        #     entries.append(cmake_cache_string("BLT_CXX_STD", "c++14", ""))
+
+        return entries
+
+    def initconfig_hardware_entries(self):
+        spec = self.spec
+        entries = super(Blttest, self).initconfig_hardware_entries()
+
+        if "+cuda" in spec:
+            entries.append(cmake_cache_option("ENABLE_CUDA", True))
+
+            # CUDA_FLAGS
+            cudaflags  = "-restrict --expt-extended-lambda "
+
+            if not spec.satisfies('cuda_arch=none'):
+                cuda_arch = spec.variants['cuda_arch'].value[0]
+                entries.append(cmake_cache_string(
+                    "CMAKE_CUDA_ARCHITECTURES",
+                    cuda_arch))
```

Comment:
I realize using `spec.satisfies()` for this check is common for RADIUSS packages, but I believe you could do something like the following:

```suggestion
            cuda_arch = spec.variants['cuda_arch'].value
            if cuda_arch != 'none':
                entries.append(cmake_cache_string(
                    "CMAKE_CUDA_ARCHITECTURES",
                    cuda_arch[0]))
```